keadm/cmd/keadm/app/cmd/edge/upgrade.go

```diff
@@ -246,14 +246,16 @@ func (up *Upgrade) Process() error {
 	// set withMqtt to false during upgrading edgecore, it will not affect the MQTT container. This is a temporary workaround and will be modified in v1.15.
 	// generate edgecore.service
 	if util.HasSystemd() {
-		err = common.GenerateServiceFile(util.KubeEdgeBinaryName, fmt.Sprintf("%s --config %s", filepath.Join(util.KubeEdgeUsrBinPath, util.KubeEdgeBinaryName), up.ConfigFilePath), false)
-		if err != nil {
+		if err := common.GenerateServiceFile(common.GenerateServiceFileOpts{
+			Process:      util.KubeEdgeBinaryName,
+			ExecStartCmd: fmt.Sprintf("%s --config %s", filepath.Join(util.KubeEdgeUsrBinPath, util.KubeEdgeBinaryName), up.ConfigFilePath),
+		}); err != nil {
 			return fmt.Errorf("failed to create edgecore.service file: %v", err)
 		}
 	}
 
 	// start new edgecore service
-	err = runEdgeCore(false)
+	err = runEdgeCore(false, "")
```

Comment:
If set `mqttImage` to empty here, when exec upgrade, the service file will be updated, and mqtt will use the default image?

Reply:
Yes, there is an issue with the upgrade itself, which does not support custom image registry.  We can fix it later when the optimization of upgrade node is finalized.